kedro/io/data_catalog.py

```diff
@@ -681,29 +681,41 @@ def add_all(
             self.add(name, dataset, replace)
 
     def add_feed_dict(self, feed_dict: dict[str, Any], replace: bool = False) -> None:
-        """Adds instances of ``MemoryDataset``, containing the data provided
-        through feed_dict.
+        """This function adds datasets to the ``DataCatalog`` using the data provided through the `feed_dict`.
+
+        `feed_dict` dictionary key is used as a dataset name, and a value is used to create an instance of
+        ``MemoryDataset`` before adding to the ``DataCatalog`` for all the value types except of ``AbstractDataset``.
+        In the last case, the ``AbstractDataset`` is added as it is.
```

Comment:
What about

```suggestion
        `feed_dict` keys are used as dataset names, and values can either be raw data or instances of ``AbstractDataset``.
        In the former case, instances of ``MemoryDataset`` are automatically created before adding to the ``DataCatalog``.
```

?

Reply:
I actually don't think mentioning `AbstractDataset` is useful here, we shouldn't really have instances of an interface right? Does the user gain anything for understanding this implementation detail?

Reply:
That's a good point. I rephrased it, but I still think it's useful to distinguish between the two cases when adding raw data and a dataset.